### ISSUE TITLE
Skipping via comment in .gitlab-ci not working for CKV_GITLABCI_2

### ISSUE THREAD

**Issue by gone-for-coding** [bridgecrewio/checkov#2986]
**Describe the issue**

The check CKV_GITLABCI_2 is giving me a false positive. So I like to skip it as below, but this is not working. I tried different variants but only `checkov --skip-check CKV_GITLABCI_2` is actually working but globally disabling the check.

**Examples**

```yaml
planOnlySubset:
  #checkov:skip=CKV_GITLABCI_2:False positive. Does not create double pipelines.
  rules:
    - if: '$CI_PIPELINE_SOURCE == "merge_request_event"'
      when: never
    - if: $CI_PIPELINE_SOURCE != "merge_request_event" && $CI_PIPELINE_SOURCE != "schedule"
      when: manual
      allow_failure: true
```
```console
Check: CKV_GITLABCI_2: "Avoid creating rules that generate double pipelines"
        FAILED for resource: .gitlab-ci.yml.*.rules.*.rules.CKV_GITLABCI_2
        File: /.gitlab-ci.yml:43-50

                43 |   rules: # checkov:skip=CKV_GITLABCI_2
                44 |     - if: '$CI_PIPELINE_SOURCE == "merge_request_event"'
                45 |       when: never
                46 |     - if: $CI_PIPELINE_SOURCE != "merge_request_event" && $CI_PIPELINE_SOURCE != "schedule"
                47 |       when: manual
                48 |       allow_failure: true
```

**Version (please complete the following information):**
 - version: 2.0.1140

**Comment by eurogig** [comment]
We'll take two approaches.  Adding skips but also fixing the FP.  Sit tight.

**Comment by Jasper-Ben** [comment]
Same issue here. Trying to skip a false positive does not work:

```
workflow:
  rules:
    # checkov:skip=CKV_GITLABCI_2: False positive. Does not create double pipelines.
    # prevent from running on gitlab.com mirror
    - if: $CI_SERVER_HOST == "gitlab.com"
      when: never
    - if: $CI_PIPELINE_SOURCE == "merge_request_event"
    - if: $CI_PIPELINE_SOURCE == "push" && $CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH
    - if: $CI_PIPELINE_SOURCE == "schedule"
    - if: $CI_PIPELINE_SOURCE == "web"
```

results in:

```
Check: CKV_GITLABCI_2: "Avoid creating rules that generate double pipelines"
        FAILED for resource: workflow.rules
        File: /.gitlab-ci.yml:14-23

                14 |     # prevent from running on gitlab.com mirror
                15 |     - if: $CI_SERVER_HOST == "gitlab.com"
                16 |       when: never
                17 |     - if: $CI_PIPELINE_SOURCE == "merge_request_event"
                18 |     - if: $CI_PIPELINE_SOURCE == "push" && $CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH
                19 |     - if: $CI_PIPELINE_SOURCE == "schedule"
                20 |     - if: $CI_PIPELINE_SOURCE == "web"
                21 |
                22 | stages:
                23 |   - setup
```

Can we reopen this issue please @eurogig @schosterbarak?

**Comment by medains** [comment]
@Jasper-Ben  Place the "# checkov:skip=CKV_GITLABCI_2" on the workflow block to skip the rule successfully.
